### ISSUE TITLE
Tests fail on Alpine Linux

### ISSUE THREAD

**Issue by Brtle** [obspy/obspy#1825]
I wanted to put ObsPy 1.0.3 in an [Alpine Linux docker](https://gist.github.com/Brtle/eb9749f2e345058c365a521319b0aad4) but a unit test fails:

```
======================================================================
ERROR: test_calc_vincenty_inverse (obspy.geodetics.tests.test_util_geodetics.UtilGeodeticsTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/obspy/geodetics/base.py", line 155, in calc_vincenty_inverse
    alpha = math.asin(sin_alpha)
ValueError: math domain error

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/obspy/geodetics/tests/test_util_geodetics.py", line 49, in test_calc_vincenty_inverse
    res = calc_vincenty_inverse(0, 0, 0, 17)
  File "/usr/lib/python3.6/site-packages/obspy/geodetics/base.py", line 191, in calc_vincenty_inverse
    raise StopIteration
StopIteration

----------------------------------------------------------------------
Ran 1386 tests in 491.079s

FAILED (errors=1)
```

This is quite weird since this unit test doesn't seem to use any particular library (except `math`). I also tried to pull the latest version of ObsPy without success.

I then find [this numpy bug](https://github.com/numpy/numpy/issues/7128). Do you think the unit test can fail due to a bug in musl's trigonometric functions like in numpy?

**Comment by krischer** [comment]
Hmm - that would be totally possible but I don't know enough about musl to be any help here. Can you maybe add a couple of print statements in the `calc_vincenty_inverse()` function - then we can compare against a non-MUSL libc and potentially narrow down the error.

**Comment by Brtle** [comment]
Hmm, I put some print statements and the [`math.atan2`](https://github.com/obspy/obspy/blob/master/obspy/geodetics/base.py#L145) doesn't give the same result.

On AlpineLinux:

```
>>> import math
>>> math.atan2(0.29237170472273677, 0.9563047559630354)
0.29670597283903605
```

On Ubuntu:

```
>>> import math
>>> math.atan2(0.29237170472273677, 0.9563047559630354)
0.2967059728390361
```

I don't think it's a problem with ObsPy, sorry for opening this issue!

**Comment by krischer** [comment]
> I don't think it's a problem with ObsPy, sorry for opening this issue!

It definitely helps to document these things so other people might be able to find it by searching for it. In any case - your example only appears to be rounded differently so I'm not sure that the `atan2` function is the issue.

**Comment by megies** [comment]
> On AlpineLinux:

```
>>> import math
>>> math.atan2(0.29237170472273677, 0.9563047559630354)
0.29670597283903605
```

> On Ubuntu:

```
>>> import math
>>> math.atan2(0.29237170472273677, 0.9563047559630354)
0.2967059728390361
```

Looks like the same just rounded away last digit?? That could even just be a thing of number of digits printed by default `float.__str__`

**Comment by Brtle** [comment]
It is not just a round issue, I still got a difference with more significant digits.

On AlpineLinux:
```
>>> print('%.25f' % math.atan2(0.29237170472273677, 0.9563047559630354))
0.2967059728390360473504472
```

On Ubuntu:
```
>>> print('%.25f' % math.atan2(0.29237170472273677, 0.9563047559630354))
0.2967059728390361028615985
```

Moreover, the test fails on [this line](https://github.com/obspy/obspy/blob/master/obspy/geodetics/base.py#L148) because the `sin_alpha` variable is equal to 1.0000000000000002220446049 and is normally in [-1, 1].

**Comment by krischer** [comment]
I see. How about clamping `sin_alpha` to [-1, 1] to make sure the operation is still defined? This should not do anything if all calculations are correct but maybe solve this particular issue? I mean its really a lot of significant digits that are still correct so accuracy should not really suffer a lot even in the degenerate case.

**Comment by Brtle** [comment]
Hmm, I absolutely don't know what this piece of code is doing. However, [this assertRaises](https://github.com/obspy/obspy/blob/master/obspy/geodetics/tests/test_util_geodetics.py#L40) is waiting a `StopIteration` which is actually a `math domain error` raised on ur [`asin` line](https://github.com/obspy/obspy/blob/master/obspy/geodetics/base.py#L148) when the `sin_alpha` is approximately equal to1.00000000000000022204.

**Comment by krischer** [comment]
@Brtle Can you try this branch here on Alpine Linux? https://github.com/krischer/obspy/tree/musl-workaround

Turns out that MUSL appears to be more correct and we rely on libc being a bit lenient.

**Comment by Brtle** [comment]
This is still failing:

```
# obspy-runtests obspy.geodetics.tests.test_util_geodetics.UtilGeodeticsTestCase.test_calc_vincenty_inverse
Running /usr/lib/python3.6/site-packages/obspy-0.0.0.dev_.g6f1a5940de-py3.6-linux-x86_64.egg/obspy/scripts/runtests.py, ObsPy version '0.0.0.dev+.g6f1a5940de'
E

======================================================================
ERROR: test_calc_vincenty_inverse (obspy.geodetics.tests.test_util_geodetics.UtilGeodeticsTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/obspy-0.0.0.dev_.g6f1a5940de-py3.6-linux-x86_64.egg/obspy/geodetics/base.py", line 149, in calc_vincenty_inverse
    alpha = math.asin(sin_alpha)
ValueError: math domain error

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/obspy-0.0.0.dev_.g6f1a5940de-py3.6-linux-x86_64.egg/obspy/geodetics/tests/test_util_geodetics.py", line 50, in test_calc_vincenty_inverse
    res = calc_vincenty_inverse(0, 0, 0, 17)
  File "/usr/lib/python3.6/site-packages/obspy-0.0.0.dev_.g6f1a5940de-py3.6-linux-x86_64.egg/obspy/geodetics/base.py", line 151, in calc_vincenty_inverse
    raise StopIteration
StopIteration

----------------------------------------------------------------------
Ran 1 test in 0.010s

FAILED (errors=1)
Do you want to report this to tests.obspy.org? [n]:
```

**Comment by krischer** [comment]
Hmm - I guess I'll have to look into running alpine linux myself. Do you happen to have the dockerfile laying around?

**Comment by Brtle** [comment]
I use [this dockerfile](https://gist.github.com/Brtle/d08c804d98e87cb00d841d95df1d5de1#file-dockerfile-musl-workaround) to test your branch. I'm a bit ashamed, this dockerfile is very ugly but it should work.

Thanks for taking time for this issue!

**Comment by megies** [comment]
Would be good to get this Dockerfile cleaned up a bit and added to our Docker setup..

**Comment by Brtle** [comment]
Fixed in #2282.

**Comment by Brtle** [comment]
Yep, I'll take a look.
